src/google/protobuf/descriptor.proto

```diff
@@ -351,6 +351,14 @@ message FileOptions {
   // generated classes from this .proto. There is no default.
   optional string objc_class_prefix = 36;
 
+  // Namespace for generated classes; defaults to the package.
+  optional string csharp_namespace = 37;
+
+  // A nested namespace for the umbrella class.  Helpful for name collisions caused by
+  // umbrella_classname conflicting with an existing type.  This will be automatically
```

Comment:
This refers to umbrella_classname which has now been removed; replace with just "the name" or something similar.
I'll try to dig into why this was required though, as coming back to it after a few years, it doesn't make much sense to me. (It might have done when "nest_classes" was false...)